File: app/src/main/java/com/faendir/lightning_launcher/multitool/proxy/Lightning.java
package com.faendir.lightning_launcher.multitool.proxy;

import android.support.annotation.Nullable;

/**
 * @author lukas
 * @since 05.07.18
 */
public interface Lightning extends Proxy {
    Event getEvent();

    void save();

    VariableSet getVariables();

    @Nullable
    Script getScriptByPathAndName(String path, String name);

    Script createScript(String path, String name, String text, int flags);

    Screen getActiveScreen();

    Script getCurrentScript();

    Script getScriptById(String id);

    void deleteScript(Script script);

    Script[] getAllScriptMatching(int flags);
}

File: app/src/main/java/com/faendir/lightning_launcher/multitool/proxy/Lightning.kt
package com.faendir.lightning_launcher.multitool.proxy

/**
 * @author lukas
 * @since 05.07.18
 */
interface Lightning : Proxy {
    val event: Event

    val variables: VariableSet

    val activeScreen: Screen

    val currentScript: Script

    fun save()

    fun getScriptByPathAndName(path: String, name: String): Script?

    fun createScript(path: String, name: String, text: String, flags: Int): Script

    fun getScriptById(id: String): Script

    fun deleteScript(script: Script)

    fun getAllScriptMatching(flags: Int): Array<Script>
}
